test/generators/typescript/preset/MarshallingPreset.spec.ts

```diff
@@ -0,0 +1,207 @@
+          }
+        }
+        return `${json.charAt(json.length-1) === ',' ? json.slice(0, json.length-1) : json}}`;
+      }
+    
+      public static unmarshal(json: string | object): NestedTest {
+        const obj = typeof json === "object" ? json : JSON.parse(json);
+        const instance = new NestedTest({});
+    
+        if (obj.stringProp !== undefined) {
+          instance.stringProp = obj.stringProp;
+        }
+    
+        //Not part of core properties
+        for (const [key, value] of Object.entries(obj).filter((([key,]) => {return !["stringProp"].includes(key);}))) {
+          //Check all pattern properties
+        
+    
+          if (instance.additionalProperties === undefined) {instance.additionalProperties = new Map();}
+          instance.additionalProperties.set(key, value as any);
+        }
+        return instance;
+      }
+    }
+    class Test {
+      private _stringProp?: string;
+      private _numberProp?: number;
+      private _objectProp?: NestedTest;
+      private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null | number>;
+      private _sTestPatternProperties?: Map<String, string>;
+    
+      constructor(input: {
+        stringProp?: string,
+        numberProp?: number,
+        objectProp?: NestedTest,
+      }) {
+        this._stringProp = input.stringProp;
+        this._numberProp = input.numberProp;
+        this._objectProp = input.objectProp;
+      }
+    
+      get stringProp(): string | undefined { return this._stringProp; }
+      set stringProp(stringProp: string | undefined) { this._stringProp = stringProp; }
+    
+      get numberProp(): number | undefined { return this._numberProp; }
+      set numberProp(numberProp: number | undefined) { this._numberProp = numberProp; }
+    
+      get objectProp(): NestedTest | undefined { return this._objectProp; }
+      set objectProp(objectProp: NestedTest | undefined) { this._objectProp = objectProp; }
+    
+      get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null | number> | undefined { return this._additionalProperties; }
+      set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null | number> | undefined) { this._additionalProperties = additionalProperties; }
+    
+      get sTestPatternProperties(): Map<String, string> | undefined { return this._sTestPatternProperties; }
+      set sTestPatternProperties(sTestPatternProperties: Map<String, string> | undefined) { this._sTestPatternProperties = sTestPatternProperties; }
+    
+      public marshal() : string {
+        let json = '{'
+        if(this.stringProp !== undefined) {
+          json += `"stringProp": ${JSON.stringify(this.stringProp)},`; 
+        }
+        if(this.numberProp !== undefined) {
+          json += `"numberProp": ${JSON.stringify(this.numberProp)},`; 
```

Comment:
`${JSON.stringify(7)}` makes this `"7"` not `7`, so you should pass numbers (and booleans) as values not as string.